tests_and_analysis/test/run_tests.py

```diff
@@ -143,7 +142,7 @@ def run_tests(pytest_options: List[str], do_report_coverage: bool,
     pytest_options = ['--import-mode=append'] + pytest_options
 
     # Start recording coverage if requested
-    cov: Union[coverage.Coverage, None] = None
+    cov: coverage.Coverage | None = None
```

Comment:
Is it still recommended this be expressed as `Optional`? Fine either way.

Reply:
I prefer to only use `Optional` to indicate an optional _parameter_. Here we have a variable that, depending on code branches, may or may not be `None`; it's a different situation, where it is not so obvious what "Optional" would mean.